src/backend/Functions/Analyze.cs

```diff
@@ -1,38 +1,50 @@
 ﻿// Copyright (c) Microsoft. All rights reserved.
 // Licensed under the MIT license. See LICENSE file in the project root for full license information.
 
-using Microsoft.Azure.WebJobs;
-using Microsoft.Azure.WebJobs.Extensions.Http;
-using Microsoft.Azure.WebJobs.Host;
-using Microsoft.Fx.Portability;
-using Microsoft.Fx.Portability.ObjectModel;
 using System;
+using System.IO;
+using System.IO.Compression;
 using System.Net;
 using System.Net.Http;
 using System.Threading.Tasks;
+using Microsoft.Azure.WebJobs;
+using Microsoft.Azure.WebJobs.Extensions.Http;
+using Microsoft.Extensions.Logging;
+using Microsoft.Fx.Portability;
+using Microsoft.Fx.Portability.ObjectModel;
+using Newtonsoft.Json;
+using WorkflowManagement;
 
 namespace Functions
 {
     public static class Analyze
     {
+        //Allows ActionFactory to be "injected" as needed, such as with a mock action factory when the function is called through tests
+        public static Func<IWorkflowActionFactory> GetActionFactory { get; set; } = () => new WorkflowActionFactory();
+
         [FunctionName("analyze")]
         public static async Task<HttpResponseMessage> Run(
             [HttpTrigger(AuthorizationLevel.Function, "post", Route = null)] HttpRequestMessage req,
-            TraceWriter log)
```

Comment:
Do we need the trace writer to see azure function output in the azure portal?

Reply:
No, the recommendation that I saw is that instead for logging, we should be using ILogger since it has better, structural logging: https://docs.microsoft.com/en-us/azure/azure-functions/functions-monitoring#write-logs-in-c-functions

Reply:
I was referring to the Logs pane in the functions dashboard. Maybe ILogger would just work. Anyway not a big deal.
![image](https://user-images.githubusercontent.com/7583839/37363693-c83be316-26b5-11e8-8ed1-56f10921a2cb.png)